File: services/core/src/main/java/ru/vsu/core/model/entity/Question.java
package ru.vsu.core.model.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document
public class Question {
    @Id
    private String questionId;
    private String name;
    private String parent;
    private Map<String, String> title;
    private Map<String, String> text;
}

File: front/admin-panel/src/entities/models/Question.ts
export type LocalizedText = Record<string, string>

type QuestionInit = Partial<Question>

export class Question {
  questionId: string
  name: string
  parent: string
  title: LocalizedText
  text: LocalizedText

  constructor(init: QuestionInit = {}) {
    this.questionId = init.questionId ?? ''
    this.name = init.name ?? ''
    this.parent = init.parent ?? ''
    this.title = init.title ?? {}
    this.text = init.text ?? {}
  }
}
